### PR TITLE
DYN-7336 Improve Unit Testing Search in Host Integrators

### PR DESCRIPTION
**Pull request by RobertGlobant20**
### Purpose

Adding unit tests for Lucene Search in Integrator Hosts
I've added a method in LuceneSearchUtility that allow to remove the current node info indexed and index new nodes information. Also I've added several unit test that will validate several test cases in all the nodes indexed in Civil3D and Revit.

### Declarations

Check these if you believe they are true

- [X] The codebase is in a better state after this PR
- [X] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [X] The level of testing this PR includes is appropriate
- [X] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB

### Release Notes

Adding unit tests for Lucene Search in Integrator Hosts

### Reviewers

@QilongTang @reddyashish

### CODE REVIEW

File: test/DynamoCoreTests/SearchModelHostIntegrationTests.cs [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        /// <summary>
+        /// Validate that the "SetParameterByName" node is found in the top 5 results from the indexed Revit 2025 nodes
+        /// </summary>
+        [Test]
+        [Category("UnitTests")]
+        public void ValidateRevitSetParameterNodeInSearch()
+        {
+            string expectedNode = "SetParameterByName";
+            string[] searchTerms = { "set parameter", "element set parameter", "setparameter", "element.setparameter" };
+            const int nodeResultsToTake = 5;
```
Comment (QilongTang) [inline test/DynamoCoreTests/SearchModelHostIntegrationTests.cs:35]: This is good but I would like to be more granular, can we record the relative rank somehow?

**Review by QilongTang — APPROVED** [review]
Thank you @RobertGlobant20 , overall this is a good step forward, I would like us to have a more granular record of the top 5 nodes ranking if possible, or maybe assert the node ranking somehow

File: test/DynamoCoreTests/SearchModelHostIntegrationTests.cs (hunk 1 above)
Comment (RobertGlobant20) [reply test/DynamoCoreTests/SearchModelHostIntegrationTests.cs:35]: @QilongTang the tests were updated so we can check the expected node in a particular position in the results.
see the commit: 956e3e50eaeee58ab0369007440c9b0c14790032

**Review by QilongTang — APPROVED** [review]

**Comment by QilongTang** [comment]
@RobertGlobant20 Please cherry-pick this change to RC3.3 branch

**Pull request merged**